cores/esp32/IPAddress.cpp

```diff
@@ -1,46 +1,48 @@
- Copyright (c) 2011 Adrian McEwen.  All right reserved.
+  IPAddress.cpp - Base class that provides IPAddress
+  Copyright (c) 2011 Adrian McEwen.  All right reserved.
 
- This library is free software; you can redistribute it and/or
- modify it under the terms of the GNU Lesser General Public
- License as published by the Free Software Foundation; either
- version 2.1 of the License, or (at your option) any later version.
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
 
- This library is distributed in the hope that it will be useful,
- but WITHOUT ANY WARRANTY; without even the implied warranty of
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- Lesser General Public License for more details.
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
 
- You should have received a copy of the GNU Lesser General Public
- License along with this library; if not, write to the Free Software
- Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
- */
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
 
-#include <Arduino.h>
-#include <IPAddress.h>
-#include <Print.h>
-#include <StreamString.h>
+#include "IPAddress.h"
+#include "Print.h"
+#include "lwip/netif.h"
+#include "StreamString.h"
 
 IPAddress::IPAddress() : IPAddress(IPv4) {}
 
 IPAddress::IPAddress(IPType ip_type)
 {
     _type = ip_type;
+    _zone = IP6_NO_ZONE;
     memset(_address.bytes, 0, sizeof(_address.bytes));
 }
 
 IPAddress::IPAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet)
 {
     _type = IPv4;
+    _zone = IP6_NO_ZONE;
     memset(_address.bytes, 0, sizeof(_address.bytes));
     _address.bytes[IPADDRESS_V4_BYTES_INDEX] = first_octet;
     _address.bytes[IPADDRESS_V4_BYTES_INDEX + 1] = second_octet;
     _address.bytes[IPADDRESS_V4_BYTES_INDEX + 2] = third_octet;
     _address.bytes[IPADDRESS_V4_BYTES_INDEX + 3] = fourth_octet;
 }
 
-IPAddress::IPAddress(uint8_t o1, uint8_t o2, uint8_t o3, uint8_t o4, uint8_t o5, uint8_t o6, uint8_t o7, uint8_t o8, uint8_t o9, uint8_t o10, uint8_t o11, uint8_t o12, uint8_t o13, uint8_t o14, uint8_t o15, uint8_t o16) {
+IPAddress::IPAddress(uint8_t o1, uint8_t o2, uint8_t o3, uint8_t o4, uint8_t o5, uint8_t o6, uint8_t o7, uint8_t o8, uint8_t o9, uint8_t o10, uint8_t o11, uint8_t o12, uint8_t o13, uint8_t o14, uint8_t o15, uint8_t o16, uint8_t z) {
```

Comment:
For compatibility, do you want to retain the constructors that don't have zone (i.e. default to 0)?

Reply:
I just wonder who would ever even use a constructor with 16 or 17 arguments?
So should we even want this kind of constructor?

If there is a constructor with 4 or 5 values tops, then I would say it is a good idea to have a default zone value.
But 16 arguments with the last one optional is really a danger to a programmer's nails, hair color and sanity (if any left...)

Reply:
in the header is where defaults are defined. The last argument defaults to `0`. Constructor comes from official Arduino core API

Reply:
I agree that a constructor with 16 arguments would rarely, if ever, get used. Anyways the linker is smart enough to remove the code if nobody uses it.